src/sorcha/utilities/sorchaConfigs.py

```diff
@@ -0,0 +1,1272 @@
+            "LINKINGFILTER": linkingfilterConfigs,
+            "OUTPUT": outputConfigs,
+            "LIGHTCURVE": lightcurveConfigs,
+            "ACTIVITY": activityConfigs,
+            "EXPERT": expertConfigs,
+        }
+        # when adding new sections in config file this general function needs the name of the section in uppercase
+        # to be the same as the attributes defined above in lowercase e.g. section INPUT has attribute input
+        # general function that reads in config file sections into there config dataclasses
+        for section, config_section in section_list.items():
+            if config_object.has_section(section):
+                extra_args = {}
+                if section == "SIMULATION":
+                    extra_args["_ephemerides_type"] = self.input.ephemerides_type
+                elif section == "FILTERS":
+                    extra_args["survey_name"] = self.survey_name
+                elif section == "SATURATION":
+                    extra_args["_observing_filters"] = self.filters.observing_filters
+                elif section == "FOV":
+                    extra_args["survey_name"] = self.survey_name
+                section_dict = dict(config_object[section])
+                config_instance = config_section(**section_dict, **extra_args)
+
+            else:
+                config_instance = config_section()  # if section not in config file take default values
+            section_key = section.lower()
+            setattr(self, section_key, config_instance)
+
+
+## below are the utility functions used to help validate the keywords, add more as needed
+
+
+def check_key_exists(value, key_name):
+    """
+    Checks to confirm that a mandatory config file value is present and has been read into the dataclass as truthy. Returns an error if value is falsy
+
+    Parameters
+    -----------
+    value : object attribute
+        value of the config file attribute
+
+    key_name : string
+        The key being checked.
+
+    Returns
+    ----------
+    None.
+
+    """
+
+    if not value:
+        logging.error(
+            f"ERROR: No value found for required key {key_name} in config file. Please check the file and try again."
+        )
+        sys.exit(
+            f"ERROR: No value found for required key {key_name} in config file. Please check the file and try again."
+        )
+
+
+def check_key_doesnt_exist(value, key_name, reason):
+    """
+    Checks to confirm that a config file value is not present and has been read into the dataclass as falsy. Returns an error if value is truthy
+
+    Parameters
+    -----------
+    value : object attribute
+        value of the config file attribute
+
+    key_name : string
+        The key being checked.
+
+    reason : string
+        reason given in the error message on why this value shouldn't be in the config file
+
+    Returns
+    ----------
+    None.
+    """
+
+    # checks to make sure value doesn't exist
+    if value:
```

Comment:
Likewise this should be:

`if value is not None:`

Basically we shouldn't be relying on Python evaluating 0 and "" as False, we should explicitly use None for unpopulated attributes.